src/ima_metrics.py

```diff
@@ -14,7 +14,11 @@ def compute_area (): #in mm2
     area += (cfg.num_matrix*11) * cfg.xbar_size * param.dac_area # 1 dac for input of f/b/d xbars, each phy xbar in d-xbar will have a dac_array, hence 8
     area += (cfg.num_matrix*4) * param.xbar_area # d-xbar has 2X xbars than f/b
     area += (cfg.num_matrix*2) * cfg.xbar_size * param.snh_area # snh for f/b xbars
-    area += cfg.num_adc * param.adc_area # adc
+    
+    #area += cfg.num_adc * param.adc_area # adc
+    for k in range (cfg.num_adc):
+        area += param.adc_area_dict[str( cfg.adc_res_new[str('matrix_adc_'+str(k))] )]        
```

Comment:
Can str(cfg.adc_res_new[str('matrix_adc_'+str(k))]) also be put within hetero-adc flag, so that in cases where num_matrix>2, this can fall back to cfg.adc_res ?

Reply:
Please look my suggestion above